# num 8 


##### № 27763 Апробация 04.03.26 (Уровень: Базовый)

Определите, сколько пятизначных чисел, записанных в семеричной системе счисления, содержат ровно одну цифру 0 и не более двух цифр 1.

In [ ]:
from itertools import product

cnt = 0
string = "0123456"
for numb in product(string, repeat=5):
    if numb.count("0") == 1 and numb.count("1") <= 2 and numb[0] != "0":
        cnt += 1
print(cnt)

Ответ: 5100

##### № 27620 Апробация 04.03.26

Все пятибуквенные слова, в составе которых могут быть только русские буквы Ц, И, Т, Р, У, С, записаны в алфавитном порядке и пронумерованы.Под каким номером в списке идёт последнее слово, которое содержит ровно две буквы И и не содержит букв Ц, стоящих рядом?

In [ ]:
from itertools import product

cnt = 0
string = "ИРСТУЦ"
for word in product(sorted(string), repeat=5):
    word = "".join(word)
    cnt += 1
    if "ЦЦ" not in word and word.count("И") == 2:
        print(cnt, word)

Ответ: 7525

##### № 27283 

Все 5-буквенные слова, составленные из букв слова ПОЛЕНИЦА, записаны в алфавитном порядке и пронумерованы. Под каким номером в списке стоит первое слово с нечётным номером, которое не начинается и не заканчивается буквой А и при этом содержит не менее трёх букв Л?

In [ ]:
from itertools import product

cnt = 0
string = sorted("ПОЛЕНИЦА")
for word in product(string, repeat=5):
    word = "".join(word)
    cnt += 1
    if word[0] != "А" and word[-1] != "А" and word.count("Л") >= 3 and cnt % 2 != 0:
        print(cnt)
        break

Ответ: 5851

##### № 29340 Открытый вариант 2026 (Уровень: Базовый)

Все шестибуквенные слова, составленные из букв А, П, Р, Е, Л, Ь, записаны в алфавитном порядке и пронумерованы.
Определите, под каким номером в этом списке стоит первое слово с нечётным номером, которое не начинается с букв А или Л и при этом содержит в своей записи не менее двух букв П.

In [5]:
from itertools import product

num = 0
string = sorted("АПРЕЛЬ")
for var in product(string, repeat=6):
    word = "".join(var)
    num += 1
    if num % 2 != 0 and word[0] not in "АЛ" and word.count("П") >= 2:
        print(num)
        break

7903


Ответ: 7903

##### № 28929 ЕГКР 18.04.26 (Уровень: Базовый)

Все пятибуквенные слова, составленные из букв С, И, М, В, О, Л, записаны в алфавитном порядке и пронумерованы.
Определите, под каким номером в этом списке стоит последнее слово с нечётным номером, которое не начинается с букв О или С и при этом содержит в своей записи ровно одну букву В и не более одной буквы С.

In [ ]:
from itertools import product

string = sorted("СОМЛИВ")
cnt = 0
for var in (lisst := list(reversed(list(product(string, repeat=5))))):
    word = "".join(var)
    cnt += 1
    if cnt % 2 == 0 and word[0] not in "ОС" and word.count("В") == 1 and word.count("С") <= 1:
        print(len(lisst) - cnt + 1)
        break

5137


Ответ: 5137

##### № 28754 Досрочная волна 2026 (Уровень: Базовый)

Все 5-буквенные слова, составленные из букв А, П, Р, Е, Л, Ь, записаны в алфавитном порядке и пронумерованы.
Определите, под каким номером в списке идёт последнее слово с чётным номером, которое не начинается с букв Ь или Р и при этом содержит в своей записи не менее двух букв Л.

In [23]:
from itertools import product

nums = []
string = sorted("АПРЕЛЬ")
cnt = 0
for var in product(string, repeat=5):
    word = "".join(var)
    # print(word)
    cnt += 1
    if cnt % 2 == 0 and word[0] not in "ЬР" and word.count("Л") >= 2:
        nums.append(cnt)
print(max(nums))

5058


Ответ: 5058

##### № 28696 (Уровень: Сложный)

Сколько существует чисел, четырнадцатеричная запись которых содержит пять цифр, среди которых есть только две нечётные цифры, причём они равны между собой и между ними стоит только одна цифра.

In [ ]:
from string import printable
from itertools import product
from re import finditer

cnt = 0
string = printable[:14]
for num in product(string, repeat=5):
    num = "".join(num)
    if num[0] != "0":
        digs = [x for x in num if int(x, 14) % 2 != 0]
        if len(digs) == 2 and len(set(digs)) == 1:
            a = [i.start() for i in finditer(digs[0], num)]
            if a[1] - a[0] == 2:
                cnt += 1
print(cnt)


[17, 25]
